githubmarkdownui/ci.py

```diff
@@ -54,12 +90,54 @@ def ci_task_list(self, status: Optional[CIStatus] = None) -> str:
 
         :param status: Only tasks with the given status will be displayed. If not given, all tasks will be displayed
         """
-        pass
+        # Want to display the results like this, and have the additional info in each line aligned with each other:
+        # <emoji>  <task name> (<task duration>)   <additional info>
+        # Need to find the longest emoji + task name + task duration and use that value to left justify each line.
+        longest_string_length = 0
+        for task in self.tasks:
+            if status and task.metadata.status != status:
+                continue
+
+            string_length = task.get_left_justified_task_string_length()
+
+            if string_length > longest_string_length:
+                longest_string_length = string_length
+
+        return code_block('\n'.join([task.build_task_string(longest_string_length) for task in self.tasks
+                                    if not status or status and task.metadata.status == status]))
+
+    def ci_job_tree(self, status: Optional[CIStatus] = None) -> str:
+        """Creates a job tree consisting of child jobs in monospaced font. Any tasks that belong to a job will show up under the job.
+
+        :param status: Only jobs and tasks with the given status will be displayed. If not given, all jobs and tasks will be
+        displayed
+        """
+        job_tree_strings = []
+
+        if status:
+            jobs_to_display = list(filter(lambda job: job.metadata.status == status, self.child_jobs))
+        else:
+            jobs_to_display = self.child_jobs
+
+        for job_index, job in enumerate(jobs_to_display):
+            # The last job should be prefixed with └─ so it looks like there's no other jobs after it.
+            if job_index == len(jobs_to_display) - 1:
+                job_tree_strings.append(f'{job.metadata.status.value} └─ {job.metadata.name}')
```

Comment:
Can we make `└─` , `├─` and `│` constants?

Reply:
it's hard thinking of good names for these, so I made them constants with these names:
`TREE_END_MARKER = └─`
`TREE_CONTINUE_MARKER = ├─`
`TREE_MORE_JOBS_MARKER = │`